internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -428,6 +440,74 @@ func (r *ClusterDeploymentsReconciler) updateInstallConfigOverrides(ctx context.
 	return nil
 }
 
+func (r *ClusterDeploymentsReconciler) syncManifests(ctx context.Context, cluster *common.Cluster, manifests map[string]string, alreadyCreatedManifests models.ListManifests) error {
+	r.Log.Debugf("Going to sync list of given with already created manifests")
+
+	for _, manifest := range alreadyCreatedManifests {
+		if _, ok := manifests[manifest.FileName]; !ok {
+			r.Log.Infof("Deleting cluster %s manifest %s", cluster.ID, manifest.FileName)
+			_ = r.Manifests.DeleteClusterManifestInternal(ctx, operations.DeleteClusterManifestParams{
+				ClusterID: *cluster.ID,
+				FileName:  manifest.FileName,
+				Folder:    swag.String(models.ManifestFolderOpenshift),
+			})
+		} else {
+			delete(manifests, manifest.FileName)
```

Comment:
Nice idea, add a comment about why you do it.
// Manifest already exists removing it from the map because later we will call `CreateClusterManifestInternal` for all manifest that are left in the map